apps/ensrainbow/src/lib/database.ts

```
@@ -178,26 +193,30 @@ export class ENSRainbowDB {
   }
 
   /**
-   * Check if an ingestion is unfinished
-   * @returns true if an ingestion is unfinished, false otherwise
+   * Get the current ingestion status
+   * @returns The current ingestion status:
+   * - null: Ingestion has never been started
+   * - "unfinished": Ingestion was started but hasn't finished
+   * - "done": Ingestion has completed successfully
    */
-  public async isIngestionUnfinished(): Promise<boolean> {
-    const value = await this.get(SYSTEM_KEY_INGESTION_UNFINISHED);
-    return value !== null;
+  public async getIngestionStatus(): Promise<string | null> {
+    return await this.get(SYSTEM_KEY_INGESTION_STATUS);
```

This function should validate that the value it reads from the db is either `null` or a recognized enum value. If the value it finds is not a recognized enum value it should throw an error.